### PR TITLE
Use nn.Container as base class for gModule

### PR DESCRIPTION
**Pull request by andreaskoepf**
Added modules to container in ctor, removed redundant methods training(),
evaluate(), share(), zeroGradParameters(), parameters(), clone() which are
now provided by the base class.

### CODE REVIEW

**Comment by soumith** [comment]
hey @andreaskoepf could you clear the merge conflicts?

**Comment by andreaskoepf** [comment]
@soumith I rebased my changes.

File: gmodule.lua [hunk 1]
```diff
@@ -118,23 +117,21 @@ function gModule:__init(inputs,outputs)
    -- computation on the graph is done through topsort of forward and backward graphs
    self.forwardnodes = self.fg:topsort()
    self.backwardnodes = self.bg:topsort()
-   self.modules = {}
```
Comment (soumith) [inline gmodule.lua]: I believe you did not mean to delete this while clearing merge conflicts.
Line 121 to 127 were added in https://github.com/torch/nngraph/pull/56/files

**Comment by andreaskoepf** [comment]
@soumith The ctor of nn.Container already initializes the member 'modules' with an empty table, see https://github.com/torch/nn/blob/master/Container.lua#L7

**Comment by andreaskoepf** [comment]
I have rebased the PR the latest changes.

**Comment by andreaskoepf** [comment]
I integrated the latest changes. Please let me know if anything is missing that prevents the integration of this PR.

**Comment by andreaskoepf** [comment]
btw gModule:map() seems to be a candidate for removal or a base class impl in nn.Container - it could simplified and  placed next to the nn.Container:share() impl.

**Comment by soumith** [comment]
Sorry for the delay. and thanks a lot for the cleanup.

**Pull request merged**